frontend/src/metabase/browse/components/BrowseApp.tsx

```diff
@@ -49,9 +45,19 @@ export const BrowseApp = ({
     return <LoadingAndErrorWrapper error />;
   }
 
+  // If no children specified, use the tab id to determine what to show inside the tab
+  if (!children) {
+    if (tab === "models") {
+      children = <BrowseModels modelsResult={modelsResult} />;
+    }
+    if (tab === "databases") {
+      children = <BrowseDatabases databasesResult={databasesResult} />;
+    }
+  }
```

Comment:
we shouldn't be reassigning component props. the other implementation was better. If you want to clean it up, make it a subcomponent

```jsx
<BrowseDataContent tab={tab}>
  {children}
</BrowseDataContent>

...

function BrowseDatacontent({ tab, children }) {

	if (children) {
	 return <>{children}</>
	}

  if (tab === "models") {
     return <BrowseModels modelsResult={modelsResult} />;
  }
    
  if (tab === "databases") {
      return <BrowseDatabases databasesResult {databasesResult} />;
  }

}


```